launcher/ui/widgets/VariableSizedImageObject.cpp

```diff
@@ -0,0 +1,98 @@
+#include "VariableSizedImageObject.h"
+
+#include <QAbstractTextDocumentLayout>
+#include <QDebug>
+#include <QPainter>
+#include <QTextObject>
+
+#include "Application.h"
+
+#include "net/NetJob.h"
+
+enum FormatProperties { ImageData = QTextFormat::UserProperty + 1 };
+
+QSizeF VariableSizedImageObject::intrinsicSize(QTextDocument* doc, int posInDocument, const QTextFormat& format)
+{
+    Q_UNUSED(posInDocument);
+
+    auto image = qvariant_cast<QImage>(format.property(ImageData));
+    auto size = image.size();
+
+    auto doc_width = doc->textWidth() - 2 * doc->documentMargin();
```

Comment:
why `- 2`?

Reply:
i added a comment for that, but just to close this, the reason is that `textWidth()` returns a width that includes the margins, so we need to remove those in order to have the image fit nicely with the rest of the text